sucuri-cleanup.py

```diff
@@ -1,22 +1,115 @@
+debug = False
+local = False
+salts = requests.get('https://api.wordpress.org/secret-key/1.1/salt/').text
 pwd = os.getcwd()
 install = pwd.split('/')[4]
-
-def salt_replace()
+install_type = pwd.split('/')[3]
+def salt_replace():
     """
     Replace the salts in the installs wp-config.php file & grab the prefix while we're there.
     """
-    wp_config = open("/nas/content/live/"+ install + "/wp-config.php", r+)
-    regex_list=["AUTH_KEY","SECURE_AUTH_KEY","LOGGED_IN_KEY","NONCE_KEY","AUTH_SALT","SECURE_AUTH_SALT","LOGGED_IN_SALT","NONCE_SALT"]
-    regex=re.compile()
-    for line in wp_config:
+    salt_regex=re.compile("^define\('(AUTH_KEY|SECURE_AUTH_KEY|LOGGED_IN_KEY|NONCE_KEY|AUTH_SALT|SECURE_AUTH_SALT|LOGGED_IN_SALT|NONCE_SALT).*$")
+    regex_prefix=re.compile("\$table_prefix\s+=\s+\'(\w+_?)\';")
+    regex_dbuser=re.compile("define\(\s+\'DB_USER\',\s+\'(\w+)\'\s+\);")
+    regex_dbpass=re.compile("define\(\s+\'DB_PASSWORD\',\s+\'([A-Za-z0-9]+)\'\s+\);")
+    regex_dbname=re.compile("define\(\s+\'DB_NAME\',\s+\'(\w+)\'\s+\);")
+    output=""
+    os.rename("./wp-config.php", "./wp-config-security_backup.php")
+    with open("./wp-config-security_backup.php", "r+") as thing:
+        text = thing.read()
+    wp_config = open("./wp-config.php", "w")
+    text=text.rstrip().split('\n')
+    for index1,line in enumerate(text):
+        if re.search(salt_regex,line) != None:
+            match=re.search(salt_regex,line).group(1)
+            if debug == True:
+                print "match= ", match
+                print "salts=", salts
+            output=re.search("define\('"+match+".*",salts).group()
+        if output != "":
+            wp_config.write(output)
+            wp_config.write("\n")
+            output=""
+            if debug == True:
+                print "Presently writing ", output, "to line ", str(index1), "at position ", wp_config.tell(), "as matched by ", re.search(salt_regex,line)
+        else:
+            if debug == True:
+                print "no match on this line! Writing ", line, "to line", str(index1), "at position", wp_config.tell()
+            wp_config.write(line)
+            wp_config.write("\n")
+        if re.search(regex_prefix,line) != None:
+            prefix = re.search(regex_prefix,line).group(1)
+        if re.search(regex_dbname,line) != None:
+            dbname = re.search(regex_dbname,line).group(1)
+        if re.search(regex_dbpass,line) != None:
+            dbpass = re.search(regex_dbpass,line).group(1)
+        if re.search(regex_dbuser,line) != None:
+            dbuser = re.search(regex_dbuser,line).group(1)
+    wp_config.close
+    return str(prefix), str(dbname), str(dbpass), str(dbuser)
+def reset_userkeys(dbvalues):
+    """
+    Reset user activation keys in the DB to prevent password stealing
+    """
+    try:
+        db=dbcon.connect("localhost",dbvalues[3],dbvalues[2],dbvalues[1])
+        db.query("update " + dbvalues[0] + "users set user_activation_key='';")
```

Comment:
Ahhhhh so I'm assumint `dbvalues[0]` is the prefix? What happens if there is no prefix? Does this still work? This might be a good thing to start a test for! When I get my `dbvalues`, and there is no prefix in the wp-config.php, what happens? What are my expectations? This allows us to ask "Ok, well this is how it should react" and challenge those assumptions like the code will down the line. Kinda like here =]